app/models/report.rb

```diff
@@ -42,124 +45,129 @@ class << self
-      sort: true, default: true, width: 100, download_default: false
+      sort: true, default: true, width: 100
     },
     {
       field: :workflow_status_ssim, label: "Errors",
       proc: ->(doc) { doc[:workflow_status_ssim].first.split("|")[2] },
-      sort: true, default: false, width: 100, download_default: false
+      sort: true, default: false, width: 100
     },
     {
       field: :file_count, label: "Files",
       proc: ->(doc) { doc[:content_file_count_itsi] },
       solr_fields: %w[content_file_count_itsi],
-      sort: false, default: true, width: 50, download_default: false
+      sort: false, default: true, width: 50
     },
     {
       field: :shelved_file_count, label: "Shelved Files",
       proc: ->(doc) { doc[:shelved_content_file_count_itsi] },
       solr_fields: %w[shelved_content_file_count_itsi],
-      sort: false, default: true, width: 50, download_default: false
+      sort: false, default: true, width: 50
     },
     {
       field: :resource_count, label: "Resources",
       proc: ->(doc) { doc[:resource_count_itsi] },
       solr_fields: %w[resource_count_itsi],
-      sort: false, default: true, width: 50, download_default: false
+      sort: false, default: true, width: 50
     },
     {
       field: :preserved_size, label: "Preservation Size",
       proc: ->(doc) { doc.preservation_size },
       solr_fields: [SolrDocument::FIELD_PRESERVATION_SIZE],
-      sort: false, default: true, width: 50, download_default: false
+      sort: false, default: true, width: 50
     },
     {
       field: :dissertation_id, label: "Dissertation ID",
       proc: ->(doc) { doc[:identifier_ssim].filter { |id| id.include?("dissertationid") }.map { |id| id.split(":").last } },
       solr_fields: %w[identifier_ssim],
-      sort: false, default: true, width: 50, download_default: false
+      sort: false, default: true, width: 50
     }
   ].freeze
 
-  COLUMN_MODEL = REPORT_FIELDS.collect do |spec|
+  COLUMN_MODEL = REPORT_FIELDS.map do |spec|
     {
-      "name" => spec[:field],
-      "jsonmap" => spec[:field],
-      "label" => spec[:label],
-      "index" => spec[:field],
-      "width" => spec[:width],
-      "sortable" => spec[:sort],
-      "hidden" => (!spec[:default])
+      "field" => spec[:field],
+      "title" => spec[:label],
+      "visible" => spec[:default],
+      "minWidth" => spec[:width],
+      # Disable sorting due to behavior that will confuse users, namely the
+      # likelihood of a user attempting to sort a column before the full report
+      # data set has loaded, namely that we are loading data sets progressively
+      # as users scroll down the reports view to keep performance satisfactory.
+      # Sorting a partial data set only to scroll to the bottom and have more
+      # rows load will look strange, and give users false confident that they
+      # are looking at the complete data set.
+      "headerSort" => false # spec[:sort]
```

Comment:
not allowing sorting makes me sad.  :-(

Reply:
It's a bummer, but Andrew requested it for the reason listed, and I can't blame him! It would be very confusing. The old UI didn't allow sorting either so nothing's lost, at least.